### ISSUE TITLE
Hyperparameters missing

### ISSUE THREAD

**Issue by malteos** [stefan-it/german-gpt2#5]
Hi @stefan-it 

thanks for sharing this model. Could you also provide the hyperparameters (learning rate, scheduler, batch size etc.) that were used for the training?

Cheers
Malte

**Comment by stefan-it** [comment]
Hi @malteos ,

unfortunately, I did that training a quite time ago, but as far as I remember, I used the default parameters from Transformers LM pre-training example and only changed the batch size. At this time it was very tricky to find a working XLA version, because I did training on a v3-8 TPU.

Best,

Stefan

**Comment by malteos** [comment]
Thanks for the reply. Do you remember to what you changed the batch size? Did you use any gradient accumulation?

**Comment by stefan-it** [comment]
Hi @malteos , I did not set any specific gradient accumulation value. I was using Transformers in version 2.10 or 2.11) and these versions do support accumulation. Default value was set to 1:

https://github.com/huggingface/transformers/blob/b42586ea560a20dcadb78472a6b4596f579e9043/src/transformers/training_args.py#L84

Batch size was between 2 or 4, I got OOMs when using the default batch size of 8:

https://github.com/huggingface/transformers/blob/b42586ea560a20dcadb78472a6b4596f579e9043/src/transformers/training_args.py#L62

**Comment by stefan-it** [comment]
Oh, I've just found the `training.args.bin` file, let's see what params are inside...

**Comment by stefan-it** [comment]
It outputs:

```json
TrainingArguments(output_dir=output-ac, overwrite_output_dir=False, do_train=True, do_eval=False, do_predict=False, evaluation_strategy=no, prediction_loss_only=False, per_device_train_batch_size=1, per_device_eval_batch_size=8, gradient_accumulation_steps=1, eval_accumulation_steps=None, learning_rate=5e-05, weight_decay=0.0, adam_beta1=0.9, adam_beta2=0.999, adam_epsilon=1e-08, max_grad_norm=1.0, num_train_epochs=5.0, max_steps=-1, lr_scheduler_type=linear, warmup_ratio=0.0, warmup_steps=0, logging_dir=runs/Jul19_18-45-27_7cc09b3bdb64, logging_strategy=steps, logging_first_step=False, logging_steps=500, save_strategy=steps, save_steps=20000, save_total_limit=None, no_cuda=False, seed=42, fp16=False, fp16_opt_level=O1, fp16_backend=auto, fp16_full_eval=False, local_rank=-1, tpu_num_cores=8, tpu_metrics_debug=False, debug=False, dataloader_drop_last=False, eval_steps=1000, dataloader_num_workers=0, past_index=-1, run_name=None, disable_tqdm=None, remove_unused_columns=True, label_names=None, load_best_model_at_end=False, metric_for_best_model=None, greater_is_better=None, ignore_data_skip=False, sharded_ddp=, deepspeed=None, label_smoothing_factor=0.0, adafactor=False, group_by_length=False, report_to=None, ddp_find_unused_parameters=None, dataloader_pin_memory=True, skip_memory_metrics=False, _n_gpu=-1)
```

So it turns out that I had to use *1* as batch size, gradient accumulation steps were *1*.

**Comment by stefan-it** [comment]
Maybe if I have some more time, I will test the new GPT neo implementation for training :)

**Comment by malteos** [comment]
Thanks for the clarification. I'll keep you posted!

**Comment by stefan-it** [comment]
Hi @malteos , I'm currently retraining this GPT-2 model with better hyperparamers.  Thanks to the awesome JAX/FLAX implementations in Transformers, training will be finished in ~80 hours, using a batch size of 128 on the full 16 GB corpus.

**Comment by malteos** [comment]
Sweet. What exact hyperparameters are you using?

**Comment by stefan-it** [comment]
I'm using this configuration here:

https://github.com/huggingface/transformers/tree/master/examples/flax/language-modeling#train-model-1

**Comment by stefan-it** [comment]
Hi @malteos ,

I've uploaded the re-trained version of German GPT-2 model! It has the same identifier as the "old" model, so using `dbmdz/german-gpt2` should be fine :hugs: 

I'm also currently training a "larger" model (training data) with 100GB of the recently released [GC4](https://german-nlp-group.github.io/projects/gc4-corpus.html) corpus.
